File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {
        let mut pre = 1;
        let mut suf = 1;
        let mut res = vec![1; nums.len()];
        for i in  0..nums.len() {
            let left = i;
            let right = (nums.len() - 1) - i;
            res[left] *= pre;
            res[right] *= suf;
            pre *= nums[left];
            suf *= nums[right];
        }
        res
    }
}

File: Data Structures & Algorithms/reverse-bits/submission-1.cpp
class Solution {
 public:
  /* int reverseBits(int n) {
    int res = n;
    for (int i = 0; i < 16; i++) {
      bool right = res & (1 << i), left = res & (1 << (31 - i));
      if (right)
        res |= (1 << (31 - i));
      else
        res &= ~(1 << (31 - i));
      if (left)
        res |= (1 << i);
      else
        res &= ~(1 << i);
    }
    return res;
  } */
  int reverseBits(int n) {
    int res = 0;
    for (int i = 0; i < 32; i++) {
      int bit = (n >> i) & 1;
      res += (bit << (31 - i));
    }
    return res;
  }
};
